internal/components/extensions_test.go

```diff
@@ -111,18 +117,75 @@ func TestDefaultExtensions(t *testing.T) {
 				return cfg
 			},
 		},
+		{
+			extension:     "awsproxy",
+			skipLifecycle: true, // Requires EC2 metadata service to be running
+		},
+		{
+			extension: "fluentbit",
+			getConfigFn: func() config.Extension {
+				cfg := extFactories["fluentbit"].CreateDefaultConfig().(*fluentbitextension.Config)
+				cfg.TCPEndpoint = "http://" + endpoint
+				return cfg
+			},
+		},
+		{
+			extension: "http_forwarder",
+			getConfigFn: func() config.Extension {
+				cfg := extFactories["http_forwarder"].CreateDefaultConfig().(*httpforwarder.Config)
+				cfg.Egress.Endpoint = "http://" + endpoint
+				cfg.Ingress.Endpoint = testutil.GetAvailableLocalAddress(t)
+				return cfg
+			},
+		},
+		{
+			extension: "oauth2client",
+			getConfigFn: func() config.Extension {
+				cfg := extFactories["oauth2client"].CreateDefaultConfig().(*oauth2clientauthextension.Config)
+				cfg.ClientID = "otel-extension"
+				cfg.ClientSecret = "testsarehard"
+				cfg.TokenURL = "http://" + endpoint
+				return cfg
+			},
+		},
+		{
+			extension:     "oidc",
+			skipLifecycle: true, // Requires a runnung OIDC server in order to complete life cycle testing
+		},
+		{
+			extension: "file_storage",
+			getConfigFn: func() config.Extension {
+				cfg := extFactories["file_storage"].CreateDefaultConfig().(*filestorage.Config)
+				cfg.Directory = "/tmp" // Need to change to testutil.NewTemporaryDirectory once merged
```

Comment:
Once https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6933 is merged, I can update this so it should work on any OS instead *nix based systems.

Reply:
That one got merged today.